src/odf/dataset/src/entities/metadata_chain.rs

```diff
@@ -399,25 +419,50 @@ pub trait MetadataChainExt: MetadataChain {
                 break;
             }
 
-            // Try to jump to the previous block with satisfaction hints taken into account
-            current_hashed_block = self
-                .get_preceding_block_with_hint(block, tail_sequence_number, merged_decision)
+            // We have processed the inclusive tail block,
+            // not overwriting current_hashed_block for potential checks.
+            if is_tail {
+                break;
+            }
```

Comment:
This fixes incorrect behavior with `options.inclusive_tail` and `!options.ignore_missing_tail`.

Otherwise, `current_hashed_block` was `None` and the condition after the loop did not work.